app/models/user.rb

```diff
@@ -141,14 +143,13 @@ def set_defaults
     @meeting_notify.nil? && @comment_notify = true
   end
 
-  OAUTH_TOKEN_URL = 'https://accounts.google.com/o/oauth2/token'
-
   def update_access_token
-    refresh_token_params = { 'refresh_token' => refresh_token,
-                             'client_id'     => nil,
-                             'client_secret' => nil,
-                             'grant_type'    => 'refresh_token' }
-    response = Net::HTTP.post_form(User::OAUTH_TOKEN_URL, refresh_token_params)
+    params = { 'refresh_token' => refresh_token,
+               'client_id'     => ENV['GOOGLE_CLIENT_ID'],
+               'client_secret' => ENV['GOOGLE_CLIENT_SECRET'],
+               'grant_type'    => 'refresh_token' }
+
+    response = Net::HTTP.post_form(URI.parse(OAUTH_TOKEN_URL), params)
```

Comment:
Actually, I can just stub this method. But i'll let the environment variables remain

Reply:
there are a couple of errors

1. `{"error"=>"invalid_request", "error_description"=>"Missing required parameter: refresh_token"}` - we don't validate the presence of `refresh_token` so this can be possible

2. `{"error"=>"invalid_request", "error_description"=>"Could not determine client ID from request."}` - if I pass in some refresh token, then this is the next error and it goes without saying the next one is for the secret

3. If I put an unrecognizable client id or secret, I get `{"error"=>"invalid_client", "error_description"=>"The OAuth client was not found."}`

4. If we ignore the above and go to L155 `new_expiration_time = Time.zone.now + decoded_response['expires_in']` then we get `NoMethodError: undefined method 'round' for nil:NilClass` because `decoded_response['expires_in']` is nil

Reply:
Are the errors you mentioned no longer appearing anymore?

Reply:
I believe they are still there. I stubbed the request so tests might not show it